quinn-proto/src/connection.rs

```diff
@@ -3538,3 +3537,5 @@ struct PathResponse {
 fn micros_from(x: Duration) -> u64 {
     x.as_secs() * 1000 * 1000 + x.subsec_micros() as u64
 }
+
+const MAX_PTO_EXPONENT: u32 = 24; // 2^24μs = ~17 seconds
```

Comment:
Given that all this is pretty much unrelated to the goal of this PR, would you mind taking care of this in a separate PR?

Reply:
It came to mind because switching to `u32` for the exponential factor in this PR makes the exponentiation considerably more likely to overflow. Admittedly it would still take long time by the wall clock, so it can wait if you prefer.

Reply:
We can do it soon, but I haven't quite grokked what you want here, so it feels inefficient for me to figure that out and do it when you can probably just crank it out. :)